File: frontend/src/types/documents/DocumentListItemDto.ts
export interface DocumentListItemDto {
  // DocumentRelatedSet
  rankInSet: number;
  roleInSetCode: string;

  // DocumentReference
  ref: number;
  typeCode: string;
  notes: string;
  isDocLocal: number;

  // LocalDocument
  label: string;
  stateCode: string;
}

File: backend/src/main/java/org/spftech/backend/dto/Document/DocumentListItemDto.java
package org.spftech.backend.dto.Document;

public record DocumentListItemDto(
    // DocumentRelatedSet
    Integer rankInSet,
    String roleInSetCode,   //to DocumentRole

    // DocumentReference,
    Long    ref,
    String  typeCode,   //to DocumentType
    String  notes,
    Short   isDocLocal,

    // LocalDocument
    String label,
    String stateCode    //to DocumentState
) {}